app/ras.py

```diff
@@ -0,0 +1,59 @@
+import json
+from jose import jwk
+from jose.utils import base64url_decode
+from social_core.backends.open_id_connect import OpenIdConnectAuth
+
+
+class RasOpenIDConnect(OpenIdConnectAuth):
+    """
+    RAS OAuth2 implementation in Python Social Auth
+    Required Backend settings are as follows:
+        SOCIAL_AUTH_RAS_KEY = '261c299a-1818-4177-a448-cb4f5602fe6d'
+        SOCIAL_AUTH_RAS_SECRET = '4de57bc2-d15d-4290-98e9-91608ed02ab5'
+    """
+
+    name = 'ras'
+    OIDC_ENDPOINT = 'https://stsstg.nih.gov'
+    EXTRA_DATA = [
+        ('expires_in', 'expires_in', True),
+        ('refresh_token', 'refresh_token', True),
+        ('id_token', 'id_token', True),
+        ('other_tokens', 'other_tokens', True),
+        ('scope', 'scope', True),
+        ('sub', 'sub', True),
+        ('ga4gh_passport_v1', 'ga4gh_passport_v1', True),
+    ]
+
+    def get_user_details(self, response):
+        # Only include passports that verify
+        algorithm = self.get_algorithm(response['id_token'])
+        response['ga4gh_passport_v1'] = [
+            passport for passport in response.get('ga4gh_passport_v1', [])
+            if self.verify_jwt(passport, algorithm)
+        ]
```

Comment:
Fetching user details is overloaded here to verify any passports returned by RAS. 

One potential gotcha here is that passports are silently discarded if they don't verify for any reason. I don't think there are really good alternatives -- raising an exception is a possibility, however that will prevent the login flow from completing if any passport fails to load.